pkg/client/client_venconn.go

```diff
@@ -0,0 +1,176 @@
+		if err != nil {
+			panic(err)
+		}
+	}()
+
+	return &VenConnClient{
+		baseURL:          baseURL,
+		agentMetadata:    agentMetadata,
+		connHandler:      handler,
+		installNamespace: installNamespace,
+		venConnName:      venConnName,
+		client:           c,
+	}, nil
+}
+
+func (c *VenConnClient) PostDataReadingsWithOptions(readings []*api.DataReading, opts Options) error {
+	return c.PostDataReadings(opts.OrgID, opts.ClusterID, readings)
+}
+
+// PostDataReadings uploads the slice of api.DataReading to the Jetstack Secure backend to be processed for later
+// viewing in the user-interface.
+func (c *VenConnClient) PostDataReadings(orgID, clusterID string, readings []*api.DataReading) error {
+	payload := api.DataReadingsPost{
+		AgentMetadata:  c.agentMetadata,
+		DataGatherTime: time.Now().UTC(),
+		DataReadings:   readings,
+	}
+	data, err := json.Marshal(payload)
+	if err != nil {
+		return err
+	}
+
+	res, err := c.Post(filepath.Join("/api/v1/org", orgID, "datareadings", clusterID), bytes.NewBuffer(data))
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	if code := res.StatusCode; code < 200 || code >= 300 {
+		errorContent := ""
+		body, err := ioutil.ReadAll(res.Body)
+		if err == nil {
+			errorContent = string(body)
+		}
+
+		return fmt.Errorf("received response with status code %d. Body: [%s]", code, errorContent)
+	}
+
+	return nil
+}
+
+// Post performs an HTTP POST request.
+func (c *VenConnClient) Post(path string, body io.Reader) (*http.Response, error) {
+	_, token, err := c.connHandler.Get(context.Background(), c.installNamespace, auth.Scope{}, types.NamespacedName{Name: c.venConnName, Namespace: c.installNamespace})
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest(http.MethodPost, fullURL(c.baseURL, path), body)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	if len(token.BearerToken) > 0 {
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token.BearerToken))
+	}
+
+	return c.client.Do(req)
+}
+
+func loadRESTConfig(path string) (*rest.Config, error) {
+	switch path {
+	// If the kubeconfig path is not provided, use the default loading rules
+	// so we read the regular KUBECONFIG variable or create a non-interactive
+	// client for agents running in cluster
+	case "":
+		loadingrules := clientcmd.NewDefaultClientConfigLoadingRules()
+		cfg, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+			loadingrules, &clientcmd.ConfigOverrides{}).ClientConfig()
+		if err != nil {
+			return nil, errors.WithStack(err)
+		}
+		return cfg, nil
+	// Otherwise use the explicitly named kubeconfig file.
+	default:
+		cfg, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+			&clientcmd.ClientConfigLoadingRules{ExplicitPath: path},
+			&clientcmd.ConfigOverrides{}).ClientConfig()
+		if err != nil {
+			return nil, errors.WithStack(err)
+		}
+		return cfg, nil
+	}
+}
```

Comment:
Self-review: I copy-pasted this from somewhere else in the Agent codebase. I'll find a better way than copy-pasting.